### DISCUSSION TITLE
Add new device class for binary-sensor software/firmware update

### DISCUSSION THREAD

**Discussion by mib1185** [home-assistant/architecture#598] [Entity Models] [answered]
# Context
Some integrations implements binary sensors to represent if an software/firmware update is available (_eq `hassio`, `shelly`, `synology_dsm`_), it would be nice to display those better.

# Proposal
Add new device class `DEVICE_CLASS_UPDATE` to binary sensor platform, which maps state on to "Update available" and state off to "up-to-date".

Also suggest to use icon `mdi:update` on frontend
![image](https://user-images.githubusercontent.com/35783820/128205796-11f2a304-9e08-4bd1-a0b9-327166a78409.png)

**current representation**
![image](https://user-images.githubusercontent.com/35783820/128206390-401fcb16-4097-4d61-a444-73a173dab3b8.png)

**suggested representation**
![image](https://user-images.githubusercontent.com/35783820/128206406-7f1fbb50-f67a-4147-b60d-d1f95c4a9b8f.png)

Furthermore this might also be seen as a preparation to #566

# Consequences
Needs to be implemented (see home-assistant/core#53945)

**Comment by thecode** [comment]
I agree, it is better to represent update binary sensors as a separate device class and not as on/off binary sensor

**Comment by frenck** [comment]
This is a fairly common sensor I've seen around in my system quite a bit. E.g., Supervisor, the updater (for core), but also with Zigbee devices.

I'm looking forward to this one 👍

**Comment by balloob** [comment]
Yeah, I guess this makes sense. 

Related topic but for another proposal: We really should implement a concept of "config" entity so we know to hide such entities from default views and show them only on device cards.

**Reply by thecode** [reply]
@balloob we should push this https://github.com/home-assistant/architecture/discussions/566

**Comment by mib1185** [comment]
ok ... there are meanwhile three strong votes, even from the boss 😎   him self, so can I assume, this proposal will be agreed? 😬

**Comment by mib1185** [comment, marked as answer]
PRs created

- home-assistant/core/pull/53945
- home-assistant/frontend/pull/9711
- home-assistant/home-assistant.io/pull/18733
- home-assistant/developers.home-assistant/pull/1013
